client/rest/handlers.go

```diff
@@ -0,0 +1,284 @@
+	writeSuccess(w, res)
+}
+
+func (k *Keys) GetKey(w http.ResponseWriter, r *http.Request) {
+	query := mux.Vars(r)
+	name := query["name"]
+	key, err := k.manager.Get(name)
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+	writeSuccess(w, &key)
+}
+
+func (k *Keys) ListKeys(w http.ResponseWriter, r *http.Request) {
+	keys, err := k.manager.List()
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+	writeSuccess(w, keys)
+}
+
+var (
+	errNonMatchingPathAndJSONKeyNames = errors.New("path and json key names don't match")
+)
+
+func (k *Keys) UpdateKey(w http.ResponseWriter, r *http.Request) {
+	uReq := new(UpdateKeyRequest)
+	if err := parseRequestJSON(r, uReq); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	query := mux.Vars(r)
+	name := query["name"]
+	if name != uReq.Name {
+		writeError(w, errNonMatchingPathAndJSONKeyNames)
+		return
+	}
+
+	if err := k.manager.Update(uReq.Name, uReq.OldPass, uReq.NewPass); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	key, err := k.manager.Get(uReq.Name)
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+	writeSuccess(w, &key)
+}
+
+func (k *Keys) DeleteKey(w http.ResponseWriter, r *http.Request) {
+	dReq := new(DeleteKeyRequest)
+	if err := parseRequestJSON(r, dReq); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	query := mux.Vars(r)
+	name := query["name"]
+	if name != dReq.Name {
+		writeError(w, errNonMatchingPathAndJSONKeyNames)
+		return
+	}
+
+	if err := k.manager.Delete(dReq.Name, dReq.Passphrase); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	resp := &ErrorResponse{Success: true}
+	writeSuccess(w, resp)
+}
+
+func (k *Keys) Register(r *mux.Router) {
+	r.HandleFunc("/keys", k.GenerateKey).Methods("POST")
+	r.HandleFunc("/keys", k.ListKeys).Methods("GET")
+	r.HandleFunc("/keys/{name}", k.GetKey).Methods("GET")
+	r.HandleFunc("/keys/{name}", k.UpdateKey).Methods("POST", "PUT")
+	r.HandleFunc("/keys/{name}", k.DeleteKey).Methods("DELETE")
+}
+
+type Context struct {
+	Keys *Keys
+}
+
+func (ctx *Context) RegisterHandlers(r *mux.Router) error {
+	ctx.Keys.Register(r)
+	r.HandleFunc("/build/send", doSend).Methods("POST")
+	r.HandleFunc("/sign", doSign).Methods("POST")
+	r.HandleFunc("/tx", doPostTx).Methods("POST")
+	r.HandleFunc("/query/account/{signature}", doAccountQuery).Methods("GET")
+
+	return nil
+}
+
+func extractAddress(signature string) (address string, err *ErrorResponse) {
+	// Expecting the signature of the form:
+	//  sig:<ADDRESS>
+	splits := strings.Split(signature, ":")
+	if len(splits) < 2 {
+		return "", &ErrorResponse{
+			Error: `expecting the signature of the form "sig:<ADDRESS>"`,
+			Code:  406,
+		}
+	}
+	if splits[0] != "sigs" {
+		return "", &ErrorResponse{
+			Error: `expecting the signature of the form "sig:<ADDRESS>"`,
+			Code:  406,
+		}
+	}
+	return splits[1], nil
+}
+
+func doAccountQuery(w http.ResponseWriter, r *http.Request) {
+	query := mux.Vars(r)
+	signature := query["signature"]
+	address, errResp := extractAddress(signature)
```

Comment:
This works, but no need to run extractAddress... ParseActor does that for you.

Not wrong, just unnecessary